sn_client/src/api.rs

```diff
@@ -38,23 +38,38 @@ use sn_protocol::{
 };
 use sn_registers::{Permissions, SignedRegister};
 use sn_transfers::{CashNote, CashNoteRedemption, MainPubkey, NanoTokens, Payment, SignedSpend};
+#[cfg(not(target_arch = "wasm32"))]
+use std::time::Instant;
 use std::{
     collections::{HashMap, HashSet},
     num::NonZeroUsize,
     path::PathBuf,
-    time::Duration,
 };
+#[cfg(not(target_arch = "wasm32"))]
 use tokio::task::spawn;
+use tokio::time::Duration;
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
to avoid having this kind of multiple target_arch dependent compliation guard across multiple files,
better to have them in a separate file, then expose ?
i.e. say we have a separate file called `target_arch_selection.rs`
then all other files can just use like: `crate::target_arch_selection::{interval, timeout, spawn, Instant}`

Reply:
ooh, nice idea.